### ISSUE TITLE
Tuning hyperparameters

### ISSUE THREAD

**Issue by sanjaythakur** [locuslab/TCN#17]
Hi,
I have a few questions about strategies to tune hyperparameters of TCN. Apparently, like any other neural network-based architecture, the performance of TCN is sensitive to the hyperparameter values whose ~ideal values vary (sometimes significantly) from task to task. It is easy to think about tuning some hyperparameters like *kernel size* and *#levels* but not others. Hence my questions are:
- Did you experiment with any normalization technique like standard normalization for TCN? Does any normalization technique help in faster convergence to the ~optimal performance? Here I am trying to get hold of tricks for TCNs similar to what exists for feed forward networks. For example, in feedforward networks, if I do a standard normalization then on most cases a learning rate of 1e-4 yields good results.
- Is there any hyperparameter search technique that works better for TCNs? Again here, I am trying to get hold of something similar to what exists for feedforward networks where usually searching in a log space yields good results. Specifically, I am wondering about the ways to tune hyperparameters like *number hidden units*, *learning rate*, and the *choice of optimizer* to use.

Thanks!!

**Comment by jerrybai1995** [comment]
Thank you for your interest in our work! Regarding your questions: 1) Yes, we did try certain normalization methods. But for some of them, such as BN, the performance eventually got worse (although we did observe faster convergence). Weightnorm works best in most of the cases, so we currently are using that in our code. 2) In general, I find that the best set of hyperparameters for TCN (especially the ones you mentioned) are pretty close to the RNN ones. Performing a grid search around the existing RNN hyperparameters (model size, learning rate, optimizer, etc.) is usually helpful.

**Comment by swilson314** [comment]
I'm sorry for asking what I expect has a really obvious answer. I downloaded the zip file and I get the following results when I try add_test.py

(base) F:\PersystCode\Python\TCN-master\TCN\adding_problem>python add_test.py
Traceback (most recent call last):
  File "add_test.py", line 5, in <module>
    from TCN.adding_problem.model import TCN
ModuleNotFoundError: No module named 'TCN'

Clearly I need to install the TCN module. I tried **>python tcn.py install**, but that didn't seem to do anything.
